### ISSUE TITLE
rubocop violations when using fastlane new_plugin

### ISSUE THREAD

**Issue by jdee** [fastlane/fastlane#8958]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [x ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

A new plugin generated using `fastlane new_plugin` does not pass rubocop without modifying .rubocop.yml.

- It wants the Fastlane-generated gemspec to be snake case (fastlane_plugin_my_new_action.gemspec).
- It doesn't like the use of `%q{}` in the Fastlane-generated action.
- It wants an empty line after magic comments (`# coding: utf-8`)

I've also seen complaints from the Eval Cop (eval is a security risk) because of the lines added to the Gemfile by `fastlane add_plugin`, but I didn't reproduce that just now.

It's not hard to disable these Cops in .rubocop.yml, but it would be nice if the `new_plugin` command
generated code that passed its own tests.

##### Complete output when running fastlane, including the stack trace and command used

```bash
[jdee@Jimmy-Dees-MacBookPro tmp]$ fastlane new_plugin my_new_action
[09:52:45]: 
Please enter a short summary of this fastlane plugin:
a short summary of this fastlane plugin
[09:52:56]: 
Please enter a detailed description of this fastlane plugin:
a detailed description of this fastlane plugin

Your plugin was successfully generated at fastlane-plugin-my_new_action/ 🚀

To get started with using this plugin, run

    fastlane add_plugin my_new_action

from a fastlane-enabled app project directory and provide the following as the path:

    /private/tmp/fastlane-plugin-my_new_action

[jdee@Jimmy-Dees-MacBookPro tmp]$ cd fastlane-plugin-my_new_action/
[jdee@Jimmy-Dees-MacBookPro fastlane-plugin-my_new_action]$ bundle install
Fetching gem metadata from https://rubygems.org/.........
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 12.0.0
Using CFPropertyList 2.3.5
Using public_suffix 2.0.5
Using ast 2.3.0
Using babosa 1.0.2
Using bundler 1.14.6
Using claide 1.0.1
Using coderay 1.1.1
Using colored 1.2
Using colored2 3.1.2
Using highline 1.7.8
Using diff-lcs 1.3
Using unf_ext 0.0.7.4
Using dotenv 2.2.0
Using excon 0.55.0
Using multipart-post 2.0.0
Using fastimage 2.1.0
Using gh_inspector 1.0.3
Using jwt 1.5.6
Using little-plugger 1.1.4
Using multi_json 1.12.1
Using memoist 0.15.0
Using os 0.9.6
Using httpclient 2.8.3
Using hurley 0.2
Using mime-types-data 3.2016.0521
Using uber 0.0.15
Using retriable 2.1.0
Using json 2.1.0
Using mini_magick 4.5.1
Using multi_xml 0.6.0
Using plist 3.2.0
Using rubyzip 1.2.1
Using security 0.1.3
Using slack-notifier 1.5.1
Using terminal-notifier 1.7.1
Using unicode-display_width 1.1.3
Using tty-screen 0.5.0
Using word_wrap 1.0.0
Using nanaimo 0.2.3
Using rouge 1.11.1
Using fastlane-plugin-my_new_action 0.1.0 from source at `.`
Using method_source 0.8.2
Using powerpack 0.1.1
Using slop 3.6.0
Using rainbow 2.2.1
Using rspec-support 3.5.0
Using ruby-progressbar 1.8.1
Using addressable 2.5.1
Using parser 2.4.0.0
Using commander-fastlane 4.4.4
Using unf 0.1.4
Installing faraday 0.12.1
Using logging 2.2.2
Using mime-types 3.1
Using representable 2.3.0
Using terminal-table 1.7.3
Using xcodeproj 1.4.4
Using xcpretty 0.2.6
Using pry 0.10.4
Using rspec-core 3.5.4
Using rspec-expectations 3.5.0
Using rspec-mocks 3.5.0
Using rubocop 0.48.1
Using domain_name 0.5.20170404
Using faraday_middleware 0.11.0.1
Using signet 0.7.3
Using xcpretty-travis-formatter 0.0.4
Using rspec 3.5.0
Using http-cookie 1.0.3
Using googleauth 0.5.1
Using faraday-cookie_jar 0.0.6
Using google-api-client 0.9.28
Using fastlane 2.28.2
Bundle complete! 7 Gemfile dependencies, 74 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
[jdee@Jimmy-Dees-MacBookPro fastlane-plugin-my_new_action]$ bundle exec rubocop
Inspecting 10 files
C.........

Offenses:

fastlane-plugin-my_new_action.gemspec:1:1: C: The name of this source file (fastlane-plugin-my_new_action.gemspec) should use snake_case.
# coding: utf-8
^
fastlane-plugin-my_new_action.gemspec:2:1: C: Add an empty line after magic comments.
lib = File.expand_path("../lib", __FILE__)
^
fastlane-plugin-my_new_action.gemspec:9:24: C: Use %q only for strings that contain both single quotes and double quotes.
  spec.author        = %q{Jimmy Dee}
                       ^^^^^^^^^^^^^
fastlane-plugin-my_new_action.gemspec:10:24: C: Use %q only for strings that contain both single quotes and double quotes.
  spec.email         = %q{jgvdthree@gmail.com}
                       ^^^^^^^^^^^^^^^^^^^^^^^
fastlane-plugin-my_new_action.gemspec:12:24: C: Use %q only for strings that contain both single quotes and double quotes.
  spec.summary       = %q{a short summary of this fastlane plugin}
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

10 files inspected, 5 offenses detected
```

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.4                                     |
| Ruby                        | 2.3.4                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.rvm/gems/ruby-2.3.4/bin/fastlane         |
| Host                        | Mac OS X 10.12.4 (16E195)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.3.4/lib                |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
lane :test do
  my_new_action
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.28.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                        | Version | Update-Status      |
| ----------------------------- | ------- | ------------------ |
| fastlane-plugin-my_new_action | 0.1.0   | 🚫 Update available |

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.0.0        |
| executable-hooks              | 1.3.2        |
| bundler-unload                | 1.0.2        |
| rubygems-bundler              | 1.4.4        |
| bundler                       | 1.14.6       |
| io-console                    | 0.4.5        |
| rake                          | 12.0.0       |
| CFPropertyList                | 2.3.5        |
| public_suffix                 | 2.0.5        |
| addressable                   | 2.5.1        |
| ast                           | 2.3.0        |
| babosa                        | 1.0.2        |
| claide                        | 1.0.1        |
| coderay                       | 1.1.1        |
| colored                       | 1.2          |
| colored2                      | 3.1.2        |
| highline                      | 1.7.8        |
| commander-fastlane            | 4.4.4        |
| diff-lcs                      | 1.3          |
| unf_ext                       | 0.0.7.4      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20170404 |
| dotenv                        | 2.2.0        |
| excon                         | 0.55.0       |
| multipart-post                | 2.0.0        |
| faraday                       | 0.12.1       |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.6        |
| faraday_middleware            | 0.11.0.1     |
| fastimage                     | 2.1.0        |
| gh_inspector                  | 1.0.3        |
| jwt                           | 1.5.6        |
| little-plugger                | 1.1.4        |
| multi_json                    | 1.12.1       |
| logging                       | 2.2.2        |
| memoist                       | 0.15.0       |
| os                            | 0.9.6        |
| signet                        | 0.7.3        |
| googleauth                    | 0.5.1        |
| httpclient                    | 2.8.3        |
| hurley                        | 0.2          |
| mime-types-data               | 3.2016.0521  |
| mime-types                    | 3.1          |
| uber                          | 0.0.15       |
| representable                 | 2.3.0        |
| retriable                     | 2.1.0        |
| google-api-client             | 0.9.28       |
| json                          | 2.1.0        |
| mini_magick                   | 4.5.1        |
| multi_xml                     | 0.6.0        |
| plist                         | 3.2.0        |
| rubyzip                       | 1.2.1        |
| security                      | 0.1.3        |
| slack-notifier                | 1.5.1        |
| terminal-notifier             | 1.7.1        |
| unicode-display_width         | 1.1.3        |
| terminal-table                | 1.7.3        |
| tty-screen                    | 0.5.0        |
| word_wrap                     | 1.0.0        |
| nanaimo                       | 0.2.3        |
| xcodeproj                     | 1.4.4        |
| rouge                         | 1.11.1       |
| xcpretty                      | 0.2.6        |
| xcpretty-travis-formatter     | 0.0.4        |
| fastlane-plugin-my_new_action | 0.1.0        |
| method_source                 | 0.8.2        |
| parser                        | 2.4.0.0      |
| powerpack                     | 0.1.1        |
| slop                          | 3.6.0        |
| pry                           | 0.10.4       |
| rainbow                       | 2.2.1        |
| rspec-support                 | 3.5.0        |
| rspec-core                    | 3.5.4        |
| rspec-expectations            | 3.5.0        |
| rspec-mocks                   | 3.5.0        |
| rspec                         | 3.5.0        |
| ruby-progressbar              | 1.8.1        |
| rubocop                       | 0.48.1       |
</details>

*generated on:* **2017-04-21**
</details>

**Comment by KrauseFx** [comment]
Nice, thanks for reporting this, and sorry for the late reply. Is that something you could prepare a PR for? It's pretty easy to get started: https://github.com/fastlane/fastlane/blob/master/YourFirstPR.md

**Comment by jdee** [comment]
@KrauseFx I can submit a PR for this if you can point me to the plugin template. The bigger problem may be that rubocop is kind of a moving target. Some automation could be useful. I'll see what I can whip up.

**Comment by KrauseFx** [comment]
Here's the template https://github.com/fastlane/fastlane/tree/master/fastlane/lib/fastlane/plugins/template 👍

**Comment by jdee** [comment]
The immediate issue is addressed by that PR ^^ (#9564). That updates rubocop in Fastlane itself to 0.49.1 and makes some changes to the gemspec template in the repo.

Regarding how to keep things in sync:

- Running `rubocop -a` immediately on plugin generation could smooth over any issues in case the generated plugin gets out of sync with the latest version of RuboCop again, though `rubocop -a` doesn't always fix everything. I got this to work nicely by adding the rubocop gem as a dependency (rather than a dev dependency) in fastlane.gemspec and then using `RuboCop::Cli` in Ruby. To avoid the additional dependency, I tried instead shelling out with `Actions.sh` and running `bundle install && bundle exec rubocop -a` in the new plugin, but that gave me trouble. That can presumably be made to work, but it might not be worth the effort.
- The version of rubocop in the generated plugin gemspec could be pinned to whatever is used in Fastlane proper. Plugin maintainers could later remove or change the limitation. Currently there is no limitation in the gemspec template.
- RuboCop is probably not scanning the .gemspec.erb file in the plugin template. I don't know if it supports ERB or if that could just be configured. That could help avoid issues in the generated code.
- If RuboCop doesn't support ERB, it would be possible to essentially run RuboCop in a new plugin in unit tests. This would also have caught the snake_case issue with the gemspec, which might be hard to catch in other ways.
